### PR TITLE
[stable/kiam] Better control of server address in agent CLI argument

### PR DESCRIPTION
**Pull request by jeevb**
#### What this PR does / why we need it:
Allow overriding server address in agent command line argument (e.g. connecting via `localhost` when the agent and server daemonsets are running on the same node group).

#### Which issue this PR fixes
*(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*
  - fixes #

#### Checklist
[Place an '[x]' (no spaces) in all applicable fields. Please remove unrelated fields.]
- [*] [DCO](https://www.helm.sh/blog/helm-dco/index.html) signed
- [*] Chart Version bumped
- [*] Variables are documented in the README.md

### CODE REVIEW

File: stable/kiam/templates/server-service.yaml [hunk 1]
```diff
@@ -1,4 +1,4 @@
-{{- if .Values.server.enabled -}}
+{{- if and .Values.server.enabled (not .Values.server.useHostNetwork) -}}
```
Comment (ewbankkit) [inline stable/kiam/templates/server-service.yaml]: Does use of host network necessarily exclude `Service`?

File: stable/kiam/templates/agent-daemonset.yaml [hunk 2]
```diff
@@ -85,7 +85,7 @@ spec:
             - --cert=/etc/kiam/tls/{{ .Values.agent.tlsCerts.certFileName }}
             - --key=/etc/kiam/tls/{{ .Values.agent.tlsCerts.keyFileName }}
             - --ca=/etc/kiam/tls/{{ .Values.agent.tlsCerts.caFileName }}
-            - --server-address={{ template "kiam.fullname" . }}-server:{{ .Values.server.service.port }}
+            - --server-address={{ if .Values.agent.serverHostOverride }}{{ .Values.agent.serverHostOverride }}{{ else if .Values.server.useHostNetwork }}localhost{{ else }}{{ template "kiam.fullname" . }}-server{{ end }}:{{ .Values.server.service.port }}
```
Comment (ewbankkit) [inline stable/kiam/templates/agent-daemonset.yaml]: Can you split over multiple lines? A single line gets unreadable.
e.g.
```
{{ if .Values.agent.serverHostOverride }}
- --server-address={{ .Values.agent.serverHostOverride }}
{{ else if .Values.server.useHostNetwork }}
...
```

File: stable/kiam/templates/server-service.yaml (hunk 1 above)
Comment (jeevb) [reply stable/kiam/templates/server-service.yaml]: I missed the use of the service for prometheus scraping. I'll fix that.

File: stable/kiam/templates/agent-daemonset.yaml (hunk 2 above)
Comment (jeevb) [reply stable/kiam/templates/agent-daemonset.yaml]: Fixed.

**Comment by jeevb** [comment]
@ewbankkit: I have addressed your comments. Does this look ok?

**Comment by jeevb** [comment]
Bump

**Pull request closed without merging**